src/Components/Tasks.jsx

```diff
@@ -0,0 +1,195 @@
+import Task from "./Task";
+import Swal from "sweetalert2";
+import Modal from "./Modal";
+
+export default function Tasks() {
+  const [dataParsed, setDataParsed] = useState([]);
+  const [toggle, setToggle] = useState(0);
+  const [token, setToken] = useState("");
+
+  console.log(token);
+
+  function showSuccessPopup()
+  {
+    Swal.fire("Good job!", "You clicked the button!", "success");
+  }
+
+
+  function getDataFromLocalStorage() {
+    const jsonFormatData = localStorage.getItem("Tasks");
+    try {
+      return JSON.parse(jsonFormatData) || [];
+    } catch {
+      return [];
+    }
+  }
+
+  function addToLocalStorage(name, assignee) {
+    let dataObject = {
+      id: `${Math.random() * 1232}`,
+      done: 0,
+      name,
+      assignee,
+      
+    };
+    dataParsed.push(dataObject);
+    setDataParsed(dataParsed);
+    let localStorageData = JSON.stringify(dataParsed);
+    localStorage.setItem("Tasks", localStorageData);
+    showSuccessPopup();
+    setTimeout(() => {
+      window.location.reload();
+    }, 1400);
+  }
+
+  function displayData() {
+    return dataParsed
+      .filter((task) => task.name.toLowerCase().includes(token))
+      .map((element, key) => {
+        return element.done == toggle ? (
+          <Task
+            id={key}
+            name={element.name}
+            assignee={element.assignee}
+            doneAttribute={element.done}
+            setItemDone={() => setItemDone(key)}
+            deleteTask={() => deleteTask(key)}
+          />
+        ) : (
+          ""
+        );
+      });
+  }
+
+  function setItemDone(id) {
+    dataParsed.find((element) => {
+      if (dataParsed.indexOf(element) == id) {
+        dataParsed[id].done = 1;
+      }
+      setDataParsed(dataParsed);
+      const localStorageData = JSON.stringify(dataParsed);
+      localStorage.setItem("Tasks", localStorageData);
+      showSuccessPopup();
+      setTimeout(() => {
+        window.location.reload();
+      }, 1400);
+    });
+  }
+
+  function deleteTask(id) {
+    Swal.fire({
+      title: "Are you sure?",
+      text: "You won't be able to revert this!",
+      icon: "warning",
+      showCancelButton: true,
+      confirmButtonColor: "#3085d6",
+      cancelButtonColor: "#d33",
+      confirmButtonText: "Yes, delete it!",
+    }).then((result) => {
+      if (result.isConfirmed) {
+        dataParsed.find((element) => {
+          if (dataParsed.indexOf(element) == id) {
+            dataParsed.splice(id, 1);
+          }
+          setDataParsed(dataParsed);
+          let localStorageData = JSON.stringify(dataParsed);
+          localStorage.setItem("Tasks", localStorageData);
+          Swal.fire("Deleted!", "Your file has been deleted.", "success");
+
+          setTimeout(() => {
+            window.location.reload();
+          }, 1400);
+        });
+      }
+    });
+  }
+
+  function renderCounter() {
+    let donesCount = 0;
+    let todoCount = 0;
+    dataParsed.map((task) => {
```

Comment:
can you rewrite this using `reduce`?

Reply:
??

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce